lighthouse-core/config/v2/config.js

```diff
@@ -0,0 +1,198 @@
+    object = Object.assign({}, object);
+    Object.keys(object).forEach(key => {
+      // We don't need to resolve objects that already have
+      if (object[key].implementation) {
+        return;
+      }
+
+      const rawPath = object[key].path || key;
+      const possiblePaths = searchPaths.map(searchPath => {
+        return path.resolve(searchPath, rawPath);
+      }).concat([
+        rawPath, // for npm plugins and absolute path usage
+        path.resolve(configPath, rawPath), // for relative config usage
+        path.resolve(process.cwd(), rawPath), // for node module usage
+      ]);
+      const resolvedPath = ConfigV2._tryResolveUntilSuccess(possiblePaths);
+      object[key] = Object.assign({}, object[key], {path: resolvedPath});
+    });
+    return object;
+  }
+
+  static objectToList(object) {
+    return Object.keys(object).reduce((list, id) => {
+      list.push(Object.assign({id}, object[id]));
+      return list;
+    }, []).sort((itemA, itemB) => itemA.order - itemB.order);
+  }
+
+  static getGathererImplementations(gatherersObject) {
+    const gathererDefinitions = ConfigV2.objectToList(gatherersObject);
+    return gathererDefinitions.map(definition => {
+      let implementation = definition.implementation;
+      if (!implementation) {
+        implementation = require(definition.path);
+      }
+
+      return Object.assign({}, definition, {implementation})
+    });
+  }
+
+  static getAuditImplementations(auditsObject) {
+    const auditDefinitions = ConfigV2.objectToList(auditsObject);
+    return auditDefinitions.map(definition => {
+      let implementation = definition.implementation;
+      if (!implementation) {
+        implementation = require(definition.path);
+      }
+
+      return Object.assign({}, definition, {implementation});
+    });
+  }
+
+  static computePasses(configJson, gatherers, audits) {
+    const gathererIds = new Set(gatherers.map(item => item.id));
+    const usedGathererIds = new Set();
+    const usedGathererNames = new Set(['traces', 'networkRecords']);
+    const requestedGathererNames = new Set(_flatten(audits.map(audit => audit.implementation.meta.requiredArtifacts)));
+    const passDefinitions = ConfigV2.objectToList(configJson.passes);
+    const passes = passDefinitions.map(definition => {
+      const foundGatherers = definition.gatherers.map(id => {
+        const gatherer = gatherers.find(item => item.id === id);
+        if (!gatherer) {
+          throw new Error(`Missing required gatherer: ${id}`);
+        }
+
+        usedGathererIds.add(id);
+        usedGathererNames.add(gatherer.implementation.name);
+        return gatherer;
+      });
+
+      return Object.assign({}, definition, {gatherers: foundGatherers});
+    });
+
+    if (gathererIds.size !== usedGathererIds.size) {
+      const unused = _subtract(gathererIds, usedGathererIds);
+      log.warn('config', `Gatherers are unused: ${unused.join(', ')}`);
```

Comment:
Seemed like a good thing to warn about in case you misname or something, this function is likely to experience the most change when we add `only`/`skip` etc, so I'd say we can re-evaluate or just remove in the filtered case.